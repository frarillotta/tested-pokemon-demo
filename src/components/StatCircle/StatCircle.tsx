import { useEffect, useState } from "react";
import CircularProgress from '@material-ui/core/CircularProgress';
import styled from "styled-components";

type StatCircleProps = {
	value: number;
	color: string;
	className: string;
}

const StatCircle = ({value, color, className}: StatCircleProps) => {

	const [progress, setProgress] = useState(0);

	useEffect(()=>{
		const timer = setInterval(()=>{
			setProgress((prevProgress) => {
				if (prevProgress < value) {
					return prevProgress + 1
				} else {
					clearTimeout(timer);
					return value;
				}
			})
		}, 1);

		return () => clearTimeout(timer);
	}, []);

	return 	<Wrapper className={className}>
		<CircularProgress 
			variant="determinate"
			style={{color: color, height: "50px", width: "50px"}}
			value={progress}
		/>
		<StatValueNumberWrapper>
			<StatValueNumber>
				{progress}
			</StatValueNumber>
		</StatValueNumberWrapper>
	</Wrapper>
	
}

const Wrapper = styled.div`
	margin: auto;
	position: relative
`


const StatValueNumberWrapper = styled.div`
	position: absolute;
	top: 0px;
	left: 0px;
	bottom: 0px;
	right: 0px;
	display: flex;
	align-items: center;
	justify-content: center;
`

const StatValueNumber = styled.div`

`

export { StatCircle }