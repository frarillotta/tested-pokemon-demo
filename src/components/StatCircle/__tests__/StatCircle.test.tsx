import { render } from "@testing-library/react";
import {StatCircle} from "../StatCircle";

describe("StatCircle", () => {
	
	describe("when loaded", () => {
		
		it("should have a consistent snapshot", () => {
			const {container} = render(<StatCircle 
				color={"blue"}
				value={45}
			/>);
			expect(container).toMatchSnapshot();
		});

	});

})