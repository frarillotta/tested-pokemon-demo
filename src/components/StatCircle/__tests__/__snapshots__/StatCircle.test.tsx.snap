// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCircle when loaded should have a consistent snapshot 1`] = `
<div>
  <div
    class="StatCircle__Wrapper-sc-1jkiqr4-0 dLMOGu"
  >
    <div
      aria-valuenow="0"
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-determinate"
      role="progressbar"
      style="width: 50px; height: 50px; transform: rotate(-90deg); color: blue;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
          style="stroke-dasharray: 126.920; stroke-dashoffset: 126.920px;"
        />
      </svg>
    </div>
    <div
      class="StatCircle__StatValueNumberWrapper-sc-1jkiqr4-1 fmUBLe"
    >
      <div
        class="StatCircle__StatValueNumber-sc-1jkiqr4-2"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
